test/tests/functional/pbs_reservations.py

```diff
@@ -1087,3 +1087,39 @@ def test_advance_resv_with_multinode_job_array(self):
         self.logger.info("Wait 180s for all the subjobs to complete")
         self.server.expect(JOB, {'job_state': 'F'},
                            id=jid2, extend='x', offset=180)
+
+    def test_reservations_with_expired_subjobs(self):
+        """
+        Test that an array job submitted to a reservation ends when
+        there are expired subjobs in the array job and job history is
+        enabled
+        """
+        self.common_steps()
+        # Submit a advance reservation and an array job to the reservation
+        # once reservation confirmed
+        now = int(time.time())
+        a = {'reserve_start': now + 20,
+             'reserve_end': now + 80}
```

Comment:
isn't 60 seconds too long? How about just 10 seconds, and you can make the sleep time of subjobs 1 second

Reply:
How about make the reservation 30 seconds.  You can start it in 10s.  Make the jobs sleep for 2s.  This means you should run through 15 of the 20 subjobs during the reservation.

Reply:
I feel 10 seconds is too small and on slower machines it may cause problems. I'll make it a reservation of 30 seconds and 2 second subjobs.

Reply:
Since the point of the test is to have a job array with some expired and some queued subjobs at the end of the reservation, then 10s should be fine.  If on slower machines the reservation is submitted and starts before the job array is submitted, then fewer subjobs will run.  They're still 2s each, so some number of them will run.  I think 10s is fine in this case.  In others, you are probably right.

Reply:
@agrawalravi90 @bhroam what's the consensus? do you want me to reduce the time further from 30 seconds to 10 and 1 second jobs?